feature: add CPU and Memory pool BDCom Switchs

kakopedreros opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #6522

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=6523) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=6523) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=6523) it.</sub>

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/30d77641-ef31-4757-99dc-bbee5f672f07) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/30d77641-ef31-4757-99dc-bbee5f672f07) completed: **No new issues**

laf commented on includes/discovery/mempools/bdcom.inc.php:
```diff
@@ -0,0 +1,36 @@
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.48.1 nmsMemoryPoolUtilization.0 = [%][This is the memory pool utilization"]
```
Please remove all the comments here, this is all in the mib `mibs/pbn/NMS-MEMORY-POOL-MIB.MIB`

kakopedreros replied:
All the comments, including the license and author info?

laf replied:
No, license and author are fine.

kakopedreros replied:
done.

laf commented on includes/discovery/mempools/bdcom.inc.php:
```diff
@@ -0,0 +1,36 @@
+if ($device['os'] == 'bdcom') {
+    echo 'BDCOM, NMS-MEMORY-POOL:';
+
+    $total = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
please use the named OID as we have the MIB available. It makes it easier to read

laf replied:
In fact you can also use snmp_get_multi or snmp_get_multi_oid in place of two gets.

kakopedreros replied:
Using the named OID i need to declare de MIB, i wich way?

This...
`snmp_get($device, 'nmsMemoryPoolTotalMemorySize.0', '-Ovq', 'NMS-MEMORY-POOL');`

or this one...?
`snmp_get($device, 'NMS-MEMORY-POOL::nmsMemoryPoolTotalMemorySize.0', '-Ovq');`

laf commented on includes/discovery/processors/bdcom.inc.php:
```diff
@@ -0,0 +1,36 @@
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.109.1.1.1.1.1.1 nmspmCPUTotalIndex.1 = [Unsigned32][Index]
```
Same comments for the comment data you have in here + using the named OID.

kakopedreros replied:
if i use the named OID, i need to declare the mib_dir in the definition os file?

laf replied:
Only if the mib dir is named differently from the os name, which it should if the vendor has multiple products so the mibs go in the vendor dir then.

laf commented on includes/polling/mempools/bdcom.inc.php:
```diff
@@ -0,0 +1,35 @@
+ * @copyright  2017 - Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+ * @author     Carlos A. Pedreros Lizama <carlos.pedreros@gmail.com>
+
+ .1.3.6.1.4.1.3320.9.48.1 nmsMemoryPoolUtilization.0 = [%][This is the memory pool utilization"]
```
Please remove comments again.

kakopedreros replied:
done.

laf commented on includes/polling/mempools/bdcom.inc.php:
```diff
@@ -0,0 +1,35 @@
+
+ */
+
+$total = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
Use named OIDs + multi get

laf commented on includes/polling/mempools/bdcom.inc.php:
```diff
@@ -0,0 +1,35 @@
+
+echo 'BDCOM Memory Pool';
+
+$mempool['total'] = snmp_get($device, '.1.3.6.1.4.1.3320.9.48.2.0', '-OvQ');
```
You seemed to have duplicated the next two snmp calls.

kakopedreros replied:
done.

laf commented:
@kakopedreros I'm not seeing any of my requests having been changed yet, let us know if you are having issues.

kakopedreros commented:
I having problems with the named OID, with the mib files located in mibs/pbn folder doesnt work.
Maybe it is the mib files in this folder are incomplete or the MIB files from BDCom are in some way differents.

[bdcom_nms_mibs.txt](https://github.com/librenms/librenms/files/968988/bdcom_nms_mibs.txt)


What do you prefere, upload the NMs mibs provided by BDCom or only use numeric OID?

laf commented:
You can upload the mib as well, it's extremely useful and easier to read by using mibs.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9dca3cb7-eeae-419e-b2f5-575dc716d391) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9dca3cb7-eeae-419e-b2f5-575dc716d391) completed: **No new issues**

laf commented:
@thanks for updating, I'll take a proper look later on but one thing I spotted was the OS logo has to be 32x32 if it's a png, can you please revert that unless you can swap it out to svg

kakopedreros commented:
Replaced the png logo with a new logo in svg. Modified the processors discovery module. Detected an error in the variable type detection.

pre-commit.php: sucess

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/08839039-6421-4b5c-a065-31c2cafdd739) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4a758e49-a5a5-4b48-b74b-2c1af94fb9b0) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/08839039-6421-4b5c-a065-31c2cafdd739) completed: **No new issues**

kakopedreros commented:
one more thing...

This was tested in 136 BDCom switches... =)
![captura de pantalla 2017-05-02 a la s 14 53 20](https://cloud.githubusercontent.com/assets/4590300/25632586/92d0f008-2f4a-11e7-8367-37aa61e6cfdc.png)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4a758e49-a5a5-4b48-b74b-2c1af94fb9b0) completed: **No new issues**

laf commented:
I've just compressed the svg down so will be good after that.
